src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -893,6 +896,12 @@ void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
             _allProfiles.Append(*profile);
         }
     }
+
+    if (_userDefaultProfileSettings)
+    {
+        // If we've loaded defaults{} we're in the "user settings" phase for sure
+        profile->Origin(OriginTag::User);
+    }

```

Comment:
@carlos-zamora this is the part that I am iffy on. I don't like it, but it is the easiest way to know that we are currently processing the user profile list compartment :|

Reply:
This crashes when profile is null -- why is profile null? i guess it's.. a first-run scenario? I'm not actually sure...

Reply:
ah, it's a profile that my DPG no longer emits